pkg/apis/kubermatic/v1/external_cluster.go

```diff
@@ -64,6 +64,124 @@ type ExternalClusterSpec struct {
+	KubeOneSpec         *ExternalClusterKubeOneSpec             `json:"kubeoneSpec,omitempty"`
+}
+
+type ExternalClusterKubeOneSpec struct {
+	SSHReference      *providerconfig.GlobalSecretKeySelector `json:"sshReference,omitempty"`
+	ManifestReference *providerconfig.GlobalSecretKeySelector `json:"manifestReference,omitempty"`
+	CloudSpec         *KubeOneCloudSpec                       `json:"cloudSpec,omitempty"`
+}
+
+type KubeOneCloudSpec struct {
+	AWS          *KubeOneAWSCloudSpec          `json:"aws,omitempty"`
+	GCP          *KubeOneGCPCloudSpec          `json:"gcp,omitempty"`
+	Azure        *KubeOneAzureCloudSpec        `json:"azure,omitempty"`
+	Digitalocean *KubeOneDigitaloceanCloudSpec `json:"digitalocean,omitempty"`
+	Openstack    *KubeOneOpenstackCloudSpec    `json:"openstack,omitempty"`
+	Packet       *KubeOnePacketCloudSpec       `json:"packet,omitempty"`
+	Hetzner      *KubeOneHetznerCloudSpec      `json:"hetzner,omitempty"`
+	VSphere      *KubeOneVSphereCloudSpec      `json:"vsphere,omitempty"`
+	Nutanix      *KubeOneNutanixCloudSpec      `json:"nutanix,omitempty"`
+}
+
+// KubeOneAWSCloudSpec specifies access data to Amazon Web Services.
+type KubeOneAWSCloudSpec struct {
+	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference"`
+
+	AccessKeyID     string `json:"accessKeyID,omitempty"`
+	SecretAccessKey string `json:"secretAccessKey,omitempty"`
+}
+
+// KubeOneGCPCloudSpec specifies access data to GCP.
+type KubeOneGCPCloudSpec struct {
+	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference"`
+
+	ServiceAccount string `json:"serviceAccount,omitempty"`
+}
+
+// KubeOneAzureCloudSpec specifies access credentials to Azure cloud.
+type KubeOneAzureCloudSpec struct {
+	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference"`
+
+	TenantID       string `json:"tenantID,omitempty"`
+	SubscriptionID string `json:"subscriptionID,omitempty"`
+	ClientID       string `json:"clientID,omitempty"`
+	ClientSecret   string `json:"clientSecret,omitempty"`
+}
+
+// KubeOneDigitaloceanCloudSpec specifies access data to DigitalOcean.
+type KubeOneDigitaloceanCloudSpec struct {
+	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference"`
+
+	Token string `json:"token,omitempty"` // Token is used to authenticate with the DigitalOcean API.
+}
+
+// KubeOneOpenstackCloudSpec specifies access data to an OpenStack cloud.
+type KubeOneOpenstackCloudSpec struct {
+	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference"`
+
+	AuthURL  string `json:"authURL,omitempty"`
+	Username string `json:"username,omitempty"`
+	Password string `json:"password,omitempty"`
+
+	// project, formally known as tenant.
+	Project string `json:"project"`
+	// project id, formally known as tenantID.
+	ProjectID string `json:"projectID"`
```

Comment:
Nit: make those comments valid Go comments.

```suggestion
	// Project, formally known as tenant.
	Project string `json:"project"`
	// ProjectID, formally known as tenantID.
	ProjectID string `json:"projectID"`
```